### PR TITLE
[SPARK-18514][DOCS] Fix the markdown for `Note:`/`NOTE:`/`Note that` across R API documentation

### PR DESCRIPTION
**Pull request by HyukjinKwon**
## What changes were proposed in this pull request?

It seems in R, there are

- `Note:`
- `NOTE:`
- `Note that`

This PR proposes to fix those to `Note:` to be consistent.

**Before**

**After**

## How was this patch tested?

The notes were found via

```bash 
grep -r "NOTE: " .
grep -r "Note that " .
```

And then fixed one by one comparing with API documentation.

After that, manually tested via `sh create-docs.sh` under `./R`.

### CODE REVIEW

File: R/pkg/R/DataFrame.R [hunk 1]
```diff
@@ -2541,7 +2541,8 @@ generateAliasesForIntersectedCols <- function (x, intersectedColNames, suffix) {
 #'
 #' Return a new SparkDataFrame containing the union of rows in this SparkDataFrame
 #' and another SparkDataFrame. This is equivalent to \code{UNION ALL} in SQL.
-#' Note that this does not remove duplicate rows across the two SparkDataFrames.
+#'
+#' Note: This does not remove duplicate rows across the two SparkDataFrames.
```
Comment (HyukjinKwon) [inline R/pkg/R/DataFrame.R]: **Before**




**After**

File: R/pkg/R/DataFrame.R [hunk 2]
```diff
@@ -2584,7 +2585,8 @@ setMethod("unionAll",
 #' Union two or more SparkDataFrames
 #'
 #' Union two or more SparkDataFrames. This is equivalent to \code{UNION ALL} in SQL.
-#' Note that this does not remove duplicate rows across the two SparkDataFrames.
+#'
+#' Note: This does not remove duplicate rows across the two SparkDataFrames.
```
Comment (HyukjinKwon) [inline R/pkg/R/DataFrame.R]: **Before**




**After**

File: R/pkg/R/functions.R [hunk 3]
```diff
@@ -2296,7 +2296,7 @@ setMethod("n", signature(x = "Column"),
 #' A pattern could be for instance \preformatted{dd.MM.yyyy} and could return a string like '18.03.1993'. All
 #' pattern letters of \code{java.text.SimpleDateFormat} can be used.
 #'
-#' NOTE: Use when ever possible specialized functions like \code{year}. These benefit from a
+#' Note: Use when ever possible specialized functions like \code{year}. These benefit from a
```
Comment (HyukjinKwon) [inline R/pkg/R/functions.R]: **Before**




**After**

File: R/pkg/R/functions.R [hunk 4]
```diff
@@ -2341,7 +2341,7 @@ setMethod("from_utc_timestamp", signature(y = "Column", x = "character"),
 #' Locate the position of the first occurrence of substr column in the given string.
 #' Returns null if either of the arguments are null.
 #'
-#' NOTE: The position is not zero based, but 1 based index. Returns 0 if substr
+#' Note: The position is not zero based, but 1 based index. Returns 0 if substr
```
Comment (HyukjinKwon) [inline R/pkg/R/functions.R]: **Before**



**After**

File: R/pkg/R/functions.R [hunk 5]
```diff
@@ -2779,7 +2779,8 @@ setMethod("window", signature(x = "Column"),
 #' locate
 #'
 #' Locate the position of the first occurrence of substr.
-#' NOTE: The position is not zero based, but 1 based index. Returns 0 if substr
+#'
+#' Note: The position is not zero based, but 1 based index. Returns 0 if substr
```
Comment (HyukjinKwon) [inline R/pkg/R/functions.R]: **Before**


**After**

**Comment by HyukjinKwon** [comment]
@felixcheung , I just noticed that the API documentation for `context.R`, `RDD.R` and `pairRDD.R` seems not appearing in the documentation. Is this true? I ended up with getting rid of them because I could not find.

**Review by srowen — APPROVED** [review]

**Comment by felixcheung** [comment]
> context.R, RDD.R and pairRDD.R

Mostly, but not all - there are a few public & documented API in context.R, for instance.
Basically if it has a `@noRd` then it will not be documented

**Comment by felixcheung** [comment]
LGTM

**Comment by HyukjinKwon** [comment]
Oh, I see. I didn't notice. Thank you both!

**Comment by srowen** [comment]
Merged to master/2.1

**Comment by HyukjinKwon** [comment]
Thank you!!!

**Pull request closed without merging**